### ISSUE TITLE
I'm not sure if externs are generated correctly for Express.js

### ISSUE THREAD

**Issue by dmitryhryppa** [haxiomic/dts2hx#13]
After achieving generated externs to work with `express.js` (see #12) I got a simple express server to work:

```haxe
final class Index {
	static function main():Void {
		final app = Express.application;
		app.listen(3000, () -> trace("Server has been started"));
	}
}
```

But, there are some things which I do not understand. 
For example, `final app = Express.application;` probably should be like `final app = Express()`, but I did not find a way to initialize it in the same way as in JavaScript or TypeScript. 
Also, there is another `Express.hx` module generated in `externs/express/express/Express.hx`:
```haxe
typedef Express = {
	@:selfCall
	function call( .... )
}
```
Which has a `@:selfCall` and which probably should work as `Express()`... But do typedefs supports this meta? 

And also, `app.get`, `app.post`, `app.use` and other functions in `express.Application` is generated as variables with types of typedefs (which also has `@:selfCall`), but probably should be generated as functions. 

Also, there is a lot of type duplicates in the externs folder. Not sure if it is bad or not:
```
externs/express/Application.hx
externs/express/express_serve_static_core/Application.hx
externs/express_serve_static_core/Application.hx
externs/serve_static/express_serve_static_core/Application.hx
```
And so on, which are absolutely the same.

**Comment by haxiomic** [comment]
Thanks for the report @dmitryhryppa, I've overhauled dependency handling which will resolve the problem where external modules are generated twice

Upgrade to 0.7.1 and let me know how you get on :)

Tomorrow I'll take a look at setting up an example using the express externs

**Comment by haxiomic** [comment]
Update: working through example, found some issues – will get back to you when they're resolved :)

**Comment by haxiomic** [comment]
Issues resolved in 0.8.0

I've created an example: [examples/express.js/Server.hx](https://github.com/haxiomic/dts2hx/blob/master/examples/express.js/Server.hx)
```haxe
/**
	Express Hello World Example
	https://expressjs.com/en/starter/hello-world.html
**/
var app = Express.call();
var port = 3000;
app.get('/', (req, res) -> res.send('Hello World!'));
app.listen(port, () -> trace('Example app listening at http://localhost:${port}'));
```

One important change from 0.6 is that the default output path is now `.haxelib/`. So you just need to add `-lib express` to your build.hxml

You mentioned the generated externs have duplicate types – in this case it's expected: express depends on express-serve-static-core, which redeclares many of the types in express to add extra fields

**Comment by dmitryhryppa** [comment]
Wow, thank you for such a quick fix! 👍

However, `get`, and some other functions are generated as `Dynamic` variables:
![image](https://user-images.githubusercontent.com/4466595/82147937-d9b12d00-9859-11ea-87f0-6b94324da40e.png)
But it probably should be something like `function get(path:String, cb:(res, req)->Void):Void; `

And `post`, `use`, `put`, `delete`, and others also variables instead of functions. I see there is a way to get completion for them:
![image](https://user-images.githubusercontent.com/4466595/82148325-272e9980-985c-11ea-8540-07630c516be5.png)
But doing it through `post.call` is not very intuitive. 

Also, if those fields are generated as variables, you can't see function overloads in the autocompletion. 

I guess there are some limitations and a lack of information about types but would be nice to work around it somehow. And I'm not pushing :) Just thoughts.

**Comment by haxiomic** [comment]
Thanks for your feedback @dmitryhryppa, I've fixed the issue with `post` and other methods in 0.9.0.

`get` is a little more tricky because it's type is `get: ((name: string) => any) & IRouterMatcher<this>;` and currently only a subset of intersections are supported. I've added a new flag `--allowIntersectionRasterization` which fixes `get`, but it's disabled by default because it causes infinite loops on some packages.

So if you convert with  `dts2hx express --allowIntersectionRasterization`, everything should work :)
